### PR TITLE
Support Variable Declaration (set initial values)

### LINKED ISSUE #8: Support for declaring a variable initial data is missing
Hello there!

There is a neat feature to set the initial value of a variable through declaration. It's done like this:

```
<<declare $bool_value = true as bool>>
//The value exists and is true
The value is {$bool_value}

<<set $bool_value = false>>
//The value is false
The value is {$bool_value}
```

Looks like this is the suggested way of using variables. While using the language server it actually marks the line with a warning if you are not declaring a variable before using it.

Warning

However, this is not supported by our runner, but looks like it should be possible [fairly easy](https://github.com/YarnSpinnerTool/YarnSpinner-Console/issues/25). Looks like the declaration data is serialized at compile time, and we are just not accesing them. 

What do you think? How can we regenerate the protobuf to support this?

### PR DESCRIPTION
**Pull request by josefalanga**
Fix Issue #8

### CODE REVIEW

File: yarnrunner_python/runner.py [hunk 1]
```diff
@@ -38,9 +38,21 @@ def __init__(self, compiled_yarn_f, names_csv_f, autostart=True, enable_tracing=
...
+    def load_initial_vars(self):
+        for key in self._compiled_yarn.initial_values:
+            initial_value = self._compiled_yarn.initial_values[key]
+            if len(initial_value.string_value) > 0:
+                self.variables[key] = initial_value.string_value
+            elif initial_value.float_value != 0:
+                self.variables[key] = initial_value.float_value
+            else:
+                self.variables[key] = initial_value.bool_value
+
```
Comment (josefalanga) [inline yarnrunner_python/runner.py:55]: this piece of code here is awful and doesn't supports initializing string variables with empty strings and initializing numeric variables with 0, and will default them to the bool_value (false).

there is a better way to identify the type of an operand?

**Comment by McJones** [comment]
Hey-oh, just saw this linked off the issue you opened on YSC.
So I can provide a bit of insight here, the issue is the Operand type is a `oneof` in protobuf so it's type isn't immediately obvious in python.
When the python is generated from this the way you determine which of the `oneof`'s the operand is is via the `HasField` method.
So to get the types back out the protobuf approved way would be something like the following:

```python
for key in self._compiled_yarn.initial_values:
    if self._compiled_yarn.initial_values[key].HasField("string_value"):
        print(key + " is a string")
    elif self._compiled_yarn.initial_values[key].HasField("bool_value"):
        print(key + " is a bool")
    elif self._compiled_yarn.initial_values[key].HasField("float_value"):
        print(key + " is a float")
```

Hope that helps

**Comment by josefalanga** [comment]
That's was a lot of help, thank you!

I knew there was a non-hacky solution, but couldn't find it myself. I looked for documentation on protobuf-python, but didn't find a damn thing. Do you have any recommended documentation?

**Comment by McJones** [comment]
I just use the protobuf dev website for looking up how `proto` elements get converted https://protobuf.dev/reference/python/python-generated/

**Review by weaversam8 — APPROVED** [review]

**Comment by weaversam8** [comment]
Well done Jose! Thanks for the assist Tim!

**Pull request merged**